From 86ddbe28eaccd385350f9a8d818a72bfe08386b1 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 25 Jan 2016 01:02:03 +0700
Subject: [PATCH 05/64] Revert "Revert "NBV: inline trivial method" and the
 revert revert that RE-BROKE something."

This reverts commit 1bc6cf5338e26dc7df1ae3632d83c68731ccd161.
---
 .../android/systemui/statusbar/phone/NavigationBarView.java    | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/NavigationBarView.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/NavigationBarView.java
index 27229eb..4cd93ea 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/NavigationBarView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/NavigationBarView.java
@@ -495,7 +495,7 @@ public class NavigationBarView extends LinearLayout {
     final Runnable mNotifyLayoutChanged = new Runnable() {
         @Override
         public void run() {
-            loadButtonArrays();
+            setupNavigationButtons(getCurrentButtonArray());
         }
     };
 
@@ -691,7 +691,7 @@ public class NavigationBarView extends LinearLayout {
                     for(int i=0;i<mButtonLayouts;i++)
                         mButtonContainerStrings[i] = Settings.System.getString(r, buttonSettings[i]);
 
-                    notifyLayoutChange(0);
+                    setupNavigationButtons(getCurrentButtonArray());
                 }};
 
             for(int i=0;i<5;i++)
@@ -773,7 +773,7 @@ public class NavigationBarView extends LinearLayout {
                 mIMEKeyArray.add(new KeyButtonInfo(actions[0], actions[1], actions[2], actions[3]));
             }
         }
-        setupNavigationButtons();
+        setupNavigationButtons(getCurrentButtonArray());
     }
 
     private ArrayList<KeyButtonInfo> getButtonsArray(final String[] userButtons) {
@@ -790,10 +790,6 @@ public class NavigationBarView extends LinearLayout {
         return mAllButtonContainers.get(mCurrentLayout);
     }
 
-    private void setupNavigationButtons() {
-        setupNavigationButtons(getCurrentButtonArray());
-    }
-
     private void setupNavigationButtons(ArrayList<KeyButtonInfo> buttonsArray) {
         final boolean stockThreeButtonLayout = buttonsArray.size() == 3;
         final int separatorSize = (int) mMenuButtonWidth;
-- 
2.5.0

